src/foundry/client/data/abstract/client-document.d.mts

```diff
@@ -692,6 +690,8 @@ declare global {
 
     interface CreateDocumentLinkOptions extends _CreateDocumentLinkOptions {}
 
+    type OnClickDocumentLinkReturn = FormApplication.Any | Promise<ApplicationV2.Any>;
```

Comment:
I know I essentially incited this signature change but good change! I was honestly thinking it'd be `MaybePromise<FormApplication.Any | ApplicationV2.Any>` and was just suggesting this for reusability, so realizing we could be more precise is nice.